tests/subprocesstest.py

```diff
@@ -0,0 +1,125 @@
+"""
+subprocesstest enables unittest test cases and suites to be run in separate
+python interpreter instances, in parallel.
+
+A base class SubprocessTestCase is provided that, when extended, will run test
+cases marked with @run_in_subprocess in a separate python interpreter.
+"""
+import unittest
+import subprocess
+import sys
+
+
+SUBPROC_TEST_ATTR = '_subproc_test'
+
+
+def run_in_subprocess(obj):
+    """
+    Marks a test case that is to be run in its own 'clean' interpreter instance.
+
+    When applied to a TestCase class, each method will be run in a separate
+    interpreter instance, in parallel.
+
+    Usage on a class::
+
+        @run_in_subprocess
+        class PatchTests(object):
```

Comment:
We should update these examples to be:

```python
from tests.subprocesstest import SubprocessTestCase, run_in_subprocess

@run_in_subprocess
class PatchTests(SubprocessTestCase):
    pass
```

Right? The only way this works is if you subclass from our `SubprocessTestCase` right?

Reply:
I wonder if we shouldn't have two base test cases here, one that assumes `@run_in_subprocess` is called on the class, and another where we have to manually mark each test case.

The above example feels verbose compared to just:

```python
from tests.subprocesstest import SubprocessTestCase


class PatchTests(SubprocessTestCase):
    pass



from tests.subprocesstest import SubprocessTestRunner, run_in_subprocess


class Tests(SubprocessTestRunner):
    @run_in_subprocess
    def test_thing(self):
        pass
```

Reply:
Hmm interesting thought. I think I'd like to side on being verbose. I imagine that for someone new to the project might gloss over the difference between `SubprocessTestRunner` and `SubprocessTestCase` when reading the tests. We could also provide the `SubprocessTestRunner` later on if we think it makes more sense.

Reply:
sounds good